src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -104,6 +104,9 @@ private async Task HandleClusterDatabaseChanged(string databaseName, long index,
                 return;
 
             using (await _disposing.ReaderLockAsync(_serverStore.ServerShutdown))
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
```

Comment:
you may keep here many opened read txs here, which isn't ideal...

Reply:
This is what is done in 6.0
https://github.com/ravendb/ravendb/blob/a17aa5aa1618a03e9f4d2675fcb4cb31bb18bad5/src/Raven.Server/Documents/DatabasesLandlord.cs#L128

Reply:
We don't use the `rawRecord` (which can be passed as a param) but there will be many read tx open.